### PR TITLE
Add warning dialogs

### PR DESCRIPTION
**Pull request by Sjoerd1993**
Adds warning dialogs if the users wants to delete a style, and if they want to reset the styles to the default values

image

image

### CODE REVIEW

File: data/ui/add_style.blp
Comment (cmkohnen) [inline data/ui/add_style.blp:1]: Probably shouldn't be part of this pr?

Comment (Sjoerd1993) [reply data/ui/add_style.blp:1]: You are correct, I based this commit on the previous one, but thought I omitted this file when making this new branch.

File: src/plot_styles.py [hunk 1]
```diff
@@ -197,6 +196,25 @@ def __init__(self, parent):
...
+        dialog = Adw.MessageDialog.new(self,
+                                       heading,
+                                       body)
+        dialog.add_response("cancel", "Cancel")
+        dialog.add_response("reset", "Reset")
+        dialog.set_close_response("cancel")
+        dialog.set_default_response("delete")
+        dialog.set_response_appearance("reset",
+                                       Adw.ResponseAppearance.DESTRUCTIVE)
+        dialog.connect("response", self.on_reset_button_press, self)
```
Comment (cmkohnen) [inline src/plot_styles.py:211]: I'd remove the self argument as this is a class function

File: src/plot_styles.py [hunk 2]
```diff
@@ -197,6 +196,25 @@ def __init__(self, parent):
...
+        dialog.present()
+
+    def on_reset_button_press(self, _, response, __):
```
Comment (cmkohnen) [inline src/plot_styles.py:214]: In conjunction with the previous comment, the last argument can be removed

File: src/plot_styles.py [hunk 3]
```diff
@@ -43,7 +43,6 @@ def reset_user_styles(self):
     for style, path in get_system_styles(self).items():
         shutil.copy(path, os.path.join(user_path, f"{style}.mplstyle"))
 
-
```
Comment (cmkohnen) [inline src/plot_styles.py:46]: Any reason for removing this line?

File: src/plot_styles.py (hunk 1 above)
Comment (Sjoerd1993) [reply src/plot_styles.py]: Yes, that was unnecessary, probably had it there out of habit. Even led to a unused argument in the relevant function.

File: src/plot_styles.py (hunk 3 above)
Comment (Sjoerd1993) [reply src/plot_styles.py]: Sharp eye, no. That was a mistake.

**Review by cmkohnen — APPROVED** [review]
